spec/tbd_tests_spec.rb

```diff
@@ -2468,54 +2488,289 @@
+      n_edges_as_heads             += 1 if edge[:type] == :headconvex
+      n_edges_as_sills             += 1 if edge[:type] == :sill
+      n_edges_as_sills             += 1 if edge[:type] == :sillconcave
+      n_edges_as_sills             += 1 if edge[:type] == :sillconvex
+      n_edges_as_jambs             += 1 if edge[:type] == :jamb
+      n_edges_as_concave_jambs     += 1 if edge[:type] == :jambconcave
+      n_edges_as_convex_jambs      += 1 if edge[:type] == :jambconvex
+      n_edges_as_doorheads         += 1 if edge[:type] == :doorhead
+      n_edges_as_doorsills         += 1 if edge[:type] == :doorsill
+      n_edges_as_doorjambs         += 1 if edge[:type] == :doorjamb
+      n_edges_as_skylightjambs     += 1 if edge[:type] == :skylightjamb
+      n_edges_as_skylightjambs     += 1 if edge[:type] == :skylightjambconvex
+      n_edges_as_corners           += 1 if edge[:type] == :corner
+      n_edges_as_concave_corners   += 1 if edge[:type] == :cornerconcave
+      n_edges_as_convex_corners    += 1 if edge[:type] == :cornerconvex
+      n_edges_as_transitions       += 1 if edge[:type] == :transition
+    end
+
+    expect(n_edges_at_grade            ).to eq( 0)
+    expect(n_edges_as_balconies        ).to eq( 2)
+    expect(n_edges_as_balconysills     ).to eq( 2) # (2x instances of GlassDoor)
+    expect(n_edges_as_concave_parapets ).to eq( 0) # 1 if parapet (not roof)
+    expect(n_edges_as_convex_parapets  ).to eq( 0) # 7 if parapet (not roof)
+    expect(n_edges_as_concave_roofs    ).to eq( 1)
+    expect(n_edges_as_convex_roofs     ).to eq( 7) # 7 - 4x "convex jambs"
+    expect(n_edges_as_rimjoists        ).to eq( 5)
+    expect(n_edges_as_concave_rimjoists).to eq( 5)
+    expect(n_edges_as_convex_rimjoists ).to eq(18)
+    expect(n_edges_as_fenestrations    ).to eq( 0)
+    expect(n_edges_as_heads            ).to eq( 2) # GlassDoor == fenestration
+    expect(n_edges_as_sills            ).to eq( 0) # (2x balconysills)
+    expect(n_edges_as_jambs            ).to eq( 4)
+    expect(n_edges_as_concave_jambs    ).to eq( 0)
+    expect(n_edges_as_convex_jambs     ).to eq( 0)
+    expect(n_edges_as_doorheads        ).to eq( 0)
+    expect(n_edges_as_doorjambs        ).to eq( 0)
+    expect(n_edges_as_doorsills        ).to eq( 0)
+    expect(n_edges_as_skylightjambs    ).to eq( 0) # (4 of 7x convex roofs)
+    expect(n_edges_as_corners          ).to eq( 0)
+    expect(n_edges_as_concave_corners  ).to eq( 4)
+    expect(n_edges_as_convex_corners   ).to eq(12)
+    expect(n_edges_as_transitions      ).to eq( 4)
+
+    #      roof PSI :  0.02 W/K•m
+    # - parapet PSI :  0.26 W/K•m
+    # ---------------------------
+    # =   delta PSI : -0.24 W/K•m
+    #
+    # e.g. East & West   : reduction of 10.4m x -0.24 W/K•m = -2.496 W/K
+    # e.g. North         : reduction of 36.6m x -0.24 W/K•m = -8.784 W/K
+    #
+    # Total length of roof/parapets : 11m + 2x 36.6m + 2x 10.4m = 105m
+    # ... 105m x -0.24 W/K•m = -25.2 W/K
```

Comment:
Tests switch from **:parapet** (default) to **:roof**. More tests needed (_tempus fugit_).